addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3305,25 +3491,80 @@ var SnippetsMenu = Widget.extend({
+                            $dropzone.addClass('invisible');
+                            const dropzoneEl = $dropzone[0];
                             $toInsert.removeClass('oe_snippet_body');
                             self.trigger_up('drop_zone_over');
+
+                            self.dragState.currentDropzoneEl = dropzoneEl;
+
+                            if (dropzoneEl.classList.contains("oe_grid_zone")) {
+                                if (!self.dragState.columnWidth || !self.dragState.columnHeight) {
+                                    const nbrColGrid = $snippet[0].getAttribute("data-oe-width-grid");
+                                    const nbrRowGrid = $snippet[0].getAttribute("data-oe-height-grid");
+                                    if (nbrColGrid || nbrRowGrid) {
+                                        // The grid dimension of snippets that
+                                        // are dynamically build (for example,
+                                        // the countdown snippet) are stored and
+                                        // does not have to be computed.
+                                        self._storeDynamicGridItemWidthAndHeight(
+                                            nbrColGrid ? parseFloat(nbrColGrid) : 12,
+                                            nbrRowGrid ? parseFloat(nbrRowGrid) : 1,
+                                            dropzoneEl.parentNode);
+                                    } else {
+                                        self._storeGridItemWidthAndHeight(dropzoneEl);
+                                    }
+                                }
+                                self._initDragOverGrid(dropzoneEl);
+                            }
                         },
                         out: function () {
-                            var prev = $toInsert.prev();
-                            if (this === prev[0]) {
-                                dropped = false;
-                                $toInsert.detach();
-                                $(this).removeClass('invisible');
-                                $toInsert.addClass('oe_snippet_body');
+                            const dropzoneEl = this;
+                            // Checking if the "out" event happens right after
+                            // the "over" of the same dropzone. If it is not the
+                            // case, we don't do anything since the previous
+                            // dropzone was already escaped (at the start of the
+                            // over).
+                            const sameDropzoneAsCurrent = self.dragState.currentDropzoneEl === dropzoneEl;
+                            if (sameDropzoneAsCurrent) {
+                                self._outOfZone(dropzoneEl);
+
+                                var prev = $toInsert.prev();
+                                let reactivateGrid = false;
+                                if (this === prev[0]) {
```

Comment:
Maybe this condition can be removed as there is already the check of sameDropzoneAsCurrent ? (Same for the same condition in SnippetEditor)